### PR TITLE
Dev/1.13.7

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

**Review by oliver-walker-hms — APPROVED** [review]

**Comment by oliver-walker-hms** [comment]
There was an issue with github actions yesterday. I think they would pass if it was working
https://www.githubstatus.com/

**Comment by alexjhawk** [comment]
> There was an issue with github actions yesterday. I think they would pass if it was working https://www.githubstatus.com/

It seems to be happening a lot, but thanks for letting me know. I want to wait until the actions pass because I've had issues with IntelliJ code formatting not catching everything since the latest update.

File: src/main/java/com/hms_networks/americas/sc/extensions/mqtt/ConstrainedMqttManager.java [hunk 1]
```diff
@@ -665,6 +674,23 @@ public void setQos(int mqttQos) throws Exception {
     }
   }
 
+  /**
+   * Sets the time interval (in secs) for sending keep alive messages on the MQTT client connection.
+   * If the MQTT manager is currently running, the MQTT connection will be updated with the new keep
```
Comment (it-hms) [inline src/main/java/com/hms_networks/americas/sc/extensions/mqtt/ConstrainedMqttManager.java:679]: Do you know if this should be set prior to start() being called? I think this comment might be misleading.  My guess is that when this option is set after the manager is already running there is no change i.e., this parameter is read when a connection is established.

**Review by it-hms — APPROVED** [review]
I approve, but I think the comment is misleading. Perhaps change the comment.

File: src/main/java/com/hms_networks/americas/sc/extensions/mqtt/ConstrainedMqttManager.java (hunk 1 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/americas/sc/extensions/mqtt/ConstrainedMqttManager.java:679]: That is a good point. The Ewon MQTT implementation does not have developer documentation, and we only have a forum post to go off of: https://techforum.ewon.biz/thread-586.html

I traced the call down through the ETK MqttClient class, and it goes all the way into a native method, which we do not have visibility in to. 

It is probably best to assume that the MqttClient will not dynamically update the keep alive interval.

I will update this PR as follows:
1. Add (optional) constructor parameter for specifying keep alive interval.
2. Make `setMqttKeepAliveIntervalSecs` trigger an MQTT restart similar to the other parameters.

**Review by it-hms — APPROVED** [review]

**Comment by alexjhawk** [comment]
@it-hms @oliver-walker-hms PR updated to improve the flow of setting the keep-alive interval. It can now be optionally set via the constructor (falls back to the default if not specified). 

I also added a commit to similarly improve the flow of configuring the MQTT thread sleep interval. It is now required to specify the MQTT thread sleep interval in the constructor. 
- Previously, it was necessary to call `setMqttThreadSleepIntervalMs()` before starting a connection. This could lead to unintended consequences (undefined sleep interval = 0) if the implementing developer forgot to make the call. This is no longer possible.

**Review by oliver-walker-hms — APPROVED** [review]

**Pull request merged**